File: 782-jewels-and-stones/jewels-and-stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        map<int,int> mp;
        for(char ch:stones){
            mp[ch]++;
        }
        int sum=0;
       for(char ch:jewels){
        if(mp.find(ch)!=mp.end()){
            sum+=mp[ch];
        }
       }
        return sum;
    }
};

File: 782-jewels-and-stones/jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        HashMap<Character,Integer> mp=new HashMap<>();
        for(char ch:stones.toCharArray()){
            mp.put(ch,mp.getOrDefault(ch,0)+1);
        }
        int sum=0;
        for(int i=0;i<jewels.length();i++){
            if(mp.containsKey(jewels.charAt(i))){
                sum+=mp.get(jewels.charAt(i));
            }
        }
        return sum;
    }
}